src/lib/settlementQueue.js

```diff
@@ -0,0 +1,139 @@
+'use strict'
+
+/**
+ * `records` is keyed by transfer_id. Each record stores a source `transfer`
+ * that has state=prepared|executed and the `payment` to which it belongs.
+ *
+ * The `transfer` arrives over a notification from the ledger, so its state is
+ * trusted. The transfers on the `payment` came from the sender, so they are not.
+ *
+ * Only when both the payment and notifications about all member source transfers
+ * have arrived can settlement proceed.
+ *
+ * TODO should `records` be stored in a database? (with an index on created_at).
+ *
+ * @param {Object} config
+ * @param {Integer} config.expiry.maxHoldTime seconds
+ */
+function SettlementQueue (config) {
+  this.maxHoldTime = config.expiry.maxHoldTime
+  this.records = {}
+}
+
+/**
+ * Called via POST /notifications (from a ledger)
+ *
+ * @param {Transfer} transfer a trusted source_transfer
+ * @returns {Payment|undefined} the trusted payment
+ */
+SettlementQueue.prototype.storeTransfer = function (transfer) {
+  if (transfer.state !== 'prepared' && transfer.state !== 'executed') {
```

Comment:
I think that a combination of autocomplete and testing means that a typo in a state string isn't a significant risk, and the constants just add unnecessary complexity. Regardless, they should either be used everywhere or nowhere, and since this isn't a refactoring PR I'll just keep using the literal strings.